### ISSUE TITLE
Application has become very slow.

### ISSUE THREAD

**Issue by scorpionhiccup** [jiahaoliuliu/WorldSpotlight#3]
I keep getting **The application may be doing too much work on its main thread.** message repeatedly. We will need to run the Async Task in the initial loading page. Also the search feature will have to be made better.

**Comment by jiahaoliuliu** [comment]
Agree. Once the new version of the app is launched, we should definitively reduce the loading time. I will put it as high priority and release a new version as soon as it is finished.
